HW18-jdbc/src/main/java/ru/otus/hw/jdbc/mapper/implementation/JdbcMapperImpl.java

```diff
@@ -0,0 +1,131 @@
+package ru.otus.hw.jdbc.mapper.implementation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.otus.hw.core.model.User;
+import ru.otus.hw.jdbc.DbExecutor;
+import ru.otus.hw.jdbc.exception.JdbcMapperException;
+import ru.otus.hw.jdbc.mapper.JdbcMapper;
+import ru.otus.hw.jdbc.sessionmanager.SessionManagerJdbc;
+
+import java.lang.reflect.InvocationTargetException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+
+public class JdbcMapperImpl<T> implements JdbcMapper<T> {
+    private static final Logger logger = LoggerFactory.getLogger(EntitySQLMetaDataImpl.class);
+
+    private final EntitySQLMetaDataImpl metaData;
+    private DbExecutor<T> dbExecutor;
+    private SessionManagerJdbc sessionManagerJdbc;
+
+    public JdbcMapperImpl(EntitySQLMetaDataImpl metaData) {
```

Comment:
Лучше использовать интерфейсы:
```suggestion
    public JdbcMapperImpl(EntitySQLMetaData metaData) {
```

И в полях:
```java
private final EntitySQLMetaData metaData;
```